extensions/pkg/terraformer/config.go

```diff
@@ -245,17 +246,29 @@ func (t *terraformer) CleanupConfiguration(ctx context.Context) error {
 	t.logger.Info("Cleaning up all terraformer configuration")
 
 	t.logger.V(1).Info("Deleting Terraform variables Secret", "name", t.variablesName)
-	if err := t.client.Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}); err != nil && !apierrors.IsNotFound(err) {
+	secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}
+	if err := t.client.Delete(ctx, secret); err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if err := extensionscontroller.DeleteAllFinalizers(ctx, t.client, secret); err != nil {
```

Comment:
Just saw this, sorry for not mentioning sooner... If the secret is not found and you call `DeleteAllFinalizers` the inner call to tryUpdate will fail as the secret does not exist. Same for the configMap and stateConfigMap.

Reply:
Yes, I should have seen that